plugins/tableselection/plugin.js

```diff
@@ -1009,6 +1011,14 @@
 						}
 
 						selection.selectRanges( ranges );
+
+						// We need to lock undoManager to consider clearing table and inserting new paragraph as single operation, and have only one undo step (#1816).
+						if ( key === 13 && editor.plugins.enterkey ) {
+							editor.undoManager.lock();
```

Comment:
It should be done by firing appropriate events (`lockSnapshot` and `unlockSnapshot`) as it's possible to use `tableselection` with `enterkey` plugin, but without `undo` one. In such case `editor.undoManager` won't be defined.

Reply:
I missed them during searching :(